### PR TITLE
HyperShift-KubeVirt: Test with OCP+CNV 4.13 on AWS

### PR DESCRIPTION
**Pull request by orenc1**
Azure and GCP cloud providers in OCP 4.12 appears to be unreliable for CI testing. For consistency and reliability, we're shifting our CI to run on AWS as an optional lane, rather than on GCP and Azure. Introducing a new env var to the `ipi-conf-aws` workflow: `COMPUTE_NODE_REPLICAS`, to control the number of worker nodes. For our testing, we will be using clusters with `m4.xlarge` master nodes and one `c5n.metal` worker node.

Signed-off-by: Oren Cohen <ocohen@redhat.com>

### CODE REVIEW

**Comment by orenc1** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

**Comment by orenc1** [comment]
/cc @davidvossel 
please take a look

**Comment by orenc1** [comment]
```
INFO[2022-12-11T22:43:38Z] Running step e2e-kubevirt-aws-ovn-deploy-cnv. 
INFO[2022-12-11T22:55:08Z] Step e2e-kubevirt-aws-ovn-deploy-cnv succeeded after 11m30s. 
INFO[2022-12-11T22:55:08Z] Running step e2e-kubevirt-aws-ovn-kv-e2e.    
INFO[2022-12-11T23:19:48Z] Step e2e-kubevirt-aws-ovn-kv-e2e succeeded after 24m40s. 
INFO[2022-12-11T23:19:48Z] Step phase test succeeded after 36m10s. 
```

**Review by davidvossel — COMMENTED** [review]
I made a few comments inline.

as next steps after we get a test that functions. I'd like to see us move the `deploy-cnv` and `kv-e2e` blocks into a workflow here [1].

The more we can move out of these yaml files in the config directory, the better. Eventually i'd like to consider moving the hypershift repo's `hack/ci/install-cnv.sh` script into a openshift ci workflow as well. 

The end result I'm shooting for here is that we don't have to do the dance between the openshift/hypershift repo and openshift/release repo as we're stabilizing CI in the future. Instead, we'll have all the logic in the openshift/release branch with each step broken into chunks chained together by a workflow.

1. https://github.com/openshift/release/tree/master/ci-operator/step-registry/hypershift/kubevirt

**Comment by davidvossel** [comment]
`INFO[2022-12-11T23:19:48Z] Step e2e-kubevirt-aws-ovn-kv-e2e succeeded after 24m40s. `

the test passed. Something is broken during the aws teardown though. 

`level=warning msg=unrecognized elastic load balancing resource type listener arn=arn:aws:elasticloadbalancing:us-east-1:460538899914:listener/net/a8024b0ab77034713a6f867282d501ff/20554edc18a25914/5955908e3c5db060`

maybe that's a transient issue though

**Comment by orenc1** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

**Comment by davidvossel** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

I want to see if this fails during deprovisioning again. I can't tell if we've hit some sort of transient issue with the aws api, or if this is legitimately permanently blocking our job from succeeding

**Comment by orenc1** [comment]
> /pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn
> 
> I want to see if this fails during deprovisioning again. I can't tell if we've hit some sort of transient issue with the aws api, or if this is legitimately permanently blocking our job from succeeding

Last run also failed in `TestCreateCluster/EnsureNoCrashingPods` test case, due to `cluster-network-operator` in the hosted control plane namespace that was restarted:
`util.go:459: Container cluster-network-operator in pod cluster-network-operator-578c544477-v62rn has a restartCount > 0 (1)`

**Comment by orenc1** [comment]
Hello @r4f4 ,
I saw you were working on a PR to fix the AWS IPI deprovisioning issue we're experiencing now:
https://github.com/openshift/installer/pull/6528
Is there a chance we can get this fix merged?
It is currently blocking the hypershift/kubevirt CI.

Thanks

**Comment by r4f4** [comment]
> Hello @r4f4 , I saw you were working on a PR to fix the AWS IPI deprovisioning issue we're experiencing now: [openshift/installer#6528](https://github.com/openshift/installer/pull/6528) Is there a chance we can get this fix merged? It is currently blocking the hypershift/kubevirt CI.

I requested review/approval from the installer team again. Hopefully they'll take a look at it soon.

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh
Comment (davidvossel) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:142]: It's possible we should return an error here that prevents the job from executing rather than clamping to a floor value of 3. I'll leave this up to the aws platform reviewer though. It doesn't matter much to me one way or the other for our purposes.

**Review by davidvossel — COMMENTED** [review]
/approve
/hold

I'm giving my approve here to unblock the hypershift changes. As long as this job remains in the main_kubevirt.yaml file, and is set to `optional:true` and `always_run: false`, i see very little risk.

Feel free to remove the hold once we've gotten to the point where the job passes the rehearsal. We'll still need more approves for the aws portion.

**Comment by orenc1** [comment]
I modified the PR to use OCP+CNV 4.13, as the AWS installer fixed is currently merged against main (4.13).

/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

**Comment by orenc1** [comment]
Hi @patrickdillon,
Could you please review the AWS-IPI step change in this PR?
We would like to introduce a new parameter to determine the number of compute nodes in the AWS cluster.

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh
Comment (r4f4) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:147]: Maybe we can simplify this to:
```suggestion
workers=${COMPUTE_NODE_REPLICAS:-3}
if [[ "${COMPUTE_NODE_REPLICAS}" -le 0 ]]; then
    workers=0
elif [[ "${COMPUTE_NODE_REPLICAS}" -gt 3 ]]; then
    workers=3
fi
```

Comment (orenc1) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh]: of course, done.

Do you think it is worth to validate in the script that the value of `$COMPUTE_NODE_REPLICAS` is a whole number? Or just let it fail later on otherwise?

**Comment by orenc1** [comment]
just to double-check after the last change:
/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

**Review by davidvossel — COMMENTED** [review]
/lgtm

**Comment by r4f4** [comment]
/approve
Install succeeded and I see 1 compute node as set in the job config.

**Comment by orenc1** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-kubevirt-e2e-kubevirt-aws-ovn

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh
Comment (patrickdillon) [inline ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:141]: why cap this at 3?

**Comment by patrickdillon** [comment]
/approve

But if there's not a good reason to cap at 3, I think we should remove the cap

File: ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh
Comment (orenc1) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh:141]: The reasoning of the cap was to prevent a case in which someone puts an unreasonably large number of compute nodes, which will incur non-negligible costs.

But if you think I should remove the cap, I'll do it.
Please let me know if I should remove it completely, or increase the cap.

Comment (orenc1) [reply ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh]: I've changed the cap to 5, I think that's a reasonable value.

**Review by nunnatsa — APPROVED** [review]
/lgtm

**Pull request merged**
